lib/tasks/publish.rake

```diff
@@ -0,0 +1,15 @@
+namespace :custom do
+  desc "This task publishes new deals and unpublishes old expired deals"
+  
+  task :publish => :environment do
+    date_today = Date.today
+    time_now = Time.current
+    Deal.find_each do |deal|
```

Comment:
@ShubhamKhandelwal7 Don't need to loop over all deals. add condition of published_at